### PR TITLE
26 delete directory button

### LINKED ISSUE #26: Fix inability to delete directories
I want to remove one entire subject from a dataset.  I am able to delete the files within the subject folder, but unable to remove the subject directory, which causes a validation failure.  should include ability to delete folders.

### PR DESCRIPTION
**Pull request by david-nishi**
Included:
- delete option added to directories (including root dir; do we want to give this option to user?)
- files are removed from apollo cache when deleted
- tests for DeleteDir and DeleteFile

resolves #26

### CODE REVIEW

File: packages/openneuro-server/graphql/resolvers/dataset.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { metadata } from './metadata.js'
 import * as dataladAnalytics from '../../datalad/analytics.js'
 import DatasetModel from '../../models/dataset.js'
 import fetch from 'node-fetch'
+import * as Sentry from '@sentry/browser'
```
Comment (nellh) [inline packages/openneuro-server/graphql/resolvers/dataset.js]: This should be @sentry/node. browser is not installed for the server, so a clean node_modules leads to a server crash.

**Review by nellh — CHANGES_REQUESTED** [review]
Looks pretty good but I haven't been able to test it end to end yet due to the Sentry module issue. Nice work on the tests.

File: packages/openneuro-server/graphql/resolvers/dataset.js (hunk 1 above)
Comment (david-nishi) [reply packages/openneuro-server/graphql/resolvers/dataset.js]: Ah thanks. The dangers of mindless copy-paste ;)

**Review by nellh — APPROVED** [review]

File: packages/openneuro-server/graphql/resolvers/dataset.js [hunk 2]
```diff
@@ -148,10 +148,9 @@ export const deleteFiles = async (
   try {
     await checkDatasetWrite(datasetId, user, userInfo)
     await Promise.all(deleteFilesTree(datasetId, fileTree))
-    await Promise.all([
-      datalad.commitFiles(datasetId, userInfo),
-      pubsub.publish('draftFilesUpdated', { datasetId }),
-    ])
+    await datalad.commitFiles(datasetId, userInfo)
+    await pubsub.publish('draftFilesUpdated', { datasetId })
+    await Promise.all([])
```
Comment (nellh) [inline packages/openneuro-server/graphql/resolvers/dataset.js]: This last line can be removed.

**Review by nellh — CHANGES_REQUESTED** [review]

**Review by nellh — APPROVED** [review]

**Pull request merged**